src/shared/acpi-fpdt.c

```diff
@@ -61,6 +61,36 @@ struct acpi_fpdt_boot {
         uint64_t exit_services_exit;
 } _packed;
 
+/* /dev/mem is deprecated on many systems try using /sys/firmware/acpi/fpdt parsing instead */
+static int acpi_get_boot_usec_kernel_parsed(usec_t *ret_loader_start, usec_t *ret_loader_exit) {
+        usec_t nsecs;
+        int r;
+
+        r = read_timestamp_file("/sys/firmware/acpi/fpdt/boot/exitbootservice_end_ns", &nsecs);
+        if (r < 0)
+                return r;
+
+        if (nsecs == 0)
+                /* Non-UEFI compatible boot. */
```

Comment:
can this actually happen? i.e. does the kernel ever expose 0 here? doesn't it suppress things already in this case?

Reply:
No, I don't think it does. But with that, I think there is another tweak needed, which is to assure that the returned times are nonzero and in order.

Let me do another push...

Reply:
I'm clear as mud here. I don't think the kernel does any value filtering, its exporting what the firmware provides. So if the firmware puts a 0 there it will show up in userspace.